server/routers/authenticate/authProviders/passkeys.ts

```diff
@@ -0,0 +1,714 @@
+          };
+        }
+      } catch (error) {
+        console.error("Authentication error:", error);
+        throw error;
+      }
+    }),
+
+  finalizePasskey: publicProcedure
+    .input(
+      z.object({
+        verificationId: z.string(),
+        email: z.string().email(),
+        sessionToken: z.string(),
+      })
+    )
+    .mutation(async ({ input, ctx }) => {
+      const { verificationId, email, sessionToken } = input;
+      const supabase = await createServerClient();
+      
+      try {
+        // Verify the session token
+        const { data: { user }, error } = await supabase.auth.getUser(sessionToken);
+        
+        if (error || !user) {
+          throw new TRPCError({
+            code: "UNAUTHORIZED",
+            message: "Invalid session token",
+          });
+        }
+        
+        // Verify the email matches
+        if (user.email !== email) {
+          throw new TRPCError({
+            code: "BAD_REQUEST",
+            message: "Email mismatch",
+          });
+        }
+        
+        // Get the credential data
+        const credentialRecord = await ctx.prisma.passkeyChallenge.findFirst({
+          where: {
+            userId: verificationId,
+            version: "credential-verification",
+          },
+        });
+        
+        if (!credentialRecord) {
+          throw new TRPCError({
+            code: "BAD_REQUEST",
+            message: "Credential record not found",
+          });
+        }
+        
+        const credentialData = JSON.parse(credentialRecord.value);
+        
+        // Use the browser's credential ID
+        const browserCredentialId = credentialData.browserCredentialId;
+
+        // Get the user profile
+        const userProfile = await ctx.prisma.userProfile.findFirst({
+          where: { supEmail: email },
+        });
+        
+        if (!userProfile) {
+          throw new TRPCError({
+            code: "NOT_FOUND",
+            message: "User profile not found",
+          });
+        }
+        
+        console.log("Creating passkey with credentialId:", browserCredentialId);
+        
+        // Create the passkey - use the browser's credential ID directly
+        await ctx.prisma.passkey.create({
+          data: {
+            userId: userProfile.supId,
+            credentialId: browserCredentialId,
+            publicKey: credentialData.credentialPublicKey,
+            signCount: credentialData.counter,
+            label: "1",
+            createdAt: new Date(),
+            lastUsedAt: new Date(),
+          },
+        });
+        
+        // Create tokens for the user
+        const accessToken = createWebAuthnAccessTokenForUser(userProfile);
+        const refreshToken = createWebAuthnRefreshTokenForUser(userProfile);
+        
+        // Set the session in Supabase
+        const { data: sessionData, error: sessionError } = await supabase.auth.setSession({
```

Comment:
Why is `supabase.auth.setSession` called both here and in `verifyAuthentication`?